### PR TITLE
Issue #654: Refactor profiles unit tests

### PR DESCRIPTION
**Pull request by katelynienaber**
## Proposed changes

Refactoring of `Profiles.unit.test.ts`

## Release Notes
Milestone: 1.7
Changelog:
- Refactored `Profiles.unit.test.ts`
- Moved profiles tests from `extension.unit.test.ts`

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] `npm run vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing **(N/A)**
- [ ] I have added integration test and it is passing **(N/A)**
- [x] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found **(N/A)**
- [ ] I have added necessary documentation (if appropriate) **(N/A)**
- [x] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

File: __mocks__/mockCreators/shared.ts [hunk 1]
```diff
@@ -42,7 +55,14 @@ export function createISessionWithoutCredentials() {
 }
 
 export function createBasicZosmfSession(profile: imperative.IProfileLoaded) {
-    return zowe.ZosmfSession.createBasicZosmfSession(profile.profile);
+    let i = zowe.ZosmfSession.createBasicZosmfSession(profile.profile);
+    return i;
```
Comment (VitGottwald) [inline __mocks__/mockCreators/shared.ts:59]: This looks like an unintended change.

**Comment by zFernand0** [comment]
LGTM! 
So close `86.36% of diff hit (target 90.00%)` :yum:

**Comment by katelynienaber** [comment]
> LGTM!
> So close `86.36% of diff hit (target 90.00%)` 😋

Oh the painnnn XD

**Comment by lauren-li** [comment]
@jellypuno Thanks for tagging us on this! We'll take a look at it and review.

**Comment by JillieBeanSim** [comment]
> LGTM! Thank you @katelynienaber
> 
> @JillieBeanSim @lauren-li This PR refactors creation of new profiles. It shouldn't affect the extenders but it would be great if you can review this as well. Just to be sure.

@jellypuno @katelynienaber functionality looks good, all tests and checks pass and creating a profile with alternative profile type works as expected.

File: src/Profiles.ts [hunk 2]
```diff
@@ -588,14 +704,14 @@ export class Profiles {
 
         // Delete from Data Set Favorites
         const favoriteDs = datasetTree.mFavorites;
```
Comment (lauren-li) [inline src/Profiles.ts:706]: I don't think favoriteDs is being used after the changes in this PR. Can we remove this declaration?

**Comment by katelynienaber** [comment]
@lauren-li @zFernand0 I removed that constant, so anyway now it should be better :3

**Comment by zFernand0** [comment]
waiting for #786

**Comment by zFernand0** [comment]
thanks for reviewing, merging soon! :yum:

**Pull request merged**